faroe/api/src/main/java/org/n52/faroe/controller/SettingDefinitionsController.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2015-2019 52°North Initiative for Geospatial Open Source
+ * Software GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.n52.faroe.controller;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.n52.faroe.Constants;
+import org.n52.faroe.SettingDefinition;
+import org.n52.faroe.SettingDefinitionGroup;
+import org.n52.faroe.dao.ServicesDao;
+import org.n52.faroe.service.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * This endpoint is responsible for fetching all the setting definitions for a a particular service
+ */
+
+@RestController
+@RequestMapping("/services/{service}/definitions")
+public class SettingDefinitionsController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SettingDefinitionsController.class);
+    private static final Gson GSON = new Gson();
+    @Inject
+    private ServicesDao servicesDao;
+
+    @GetMapping
+    public ResponseEntity<Object> getSettingDefinitons(@PathVariable("service") String serviceName) {
+        LOG.info("Getting definitions");
+        try {
+            final Service service = servicesDao.getServiceByName(serviceName);
+            final Set<SettingDefinition<?>> settingDefinitionSet = service.getSettingsService()
+                    .getSettingDefinitions();
+            final Map<String, Object> response = new HashMap<>();
+            response.put(Constants.NAME, service.getName());
+            response.put(Constants.SETTING_DEFINITIONS, GSON.toJson(settingDefinitionSet));
+            return new ResponseEntity<>(response, HttpStatus.OK);
+        } catch (Exception e) {
+            LOG.error(
+                    String.format("Couldn't fetch setting definitions for the service: %s", serviceName));
+            e.printStackTrace();
```

Comment:
Don't print the stack trace to `stdout`, include it in the `LOG.error()` statement. Also `LOG.error()` can format strings...